package/kedro_viz/api/responses.py

```diff
@@ -0,0 +1,199 @@
+                "modular_pipelines": [],
+                "type": "task",
+                "parameters": {
+                    "test_size": 0.2,
+                    "random_state": 3,
+                    "features": [
+                        "engines",
+                        "passenger_capacity",
+                        "crew",
+                        "d_check_complete",
+                        "moon_clearance_complete",
+                        "iata_approved",
+                        "company_rating",
+                        "review_scores_rating",
+                    ],
+                },
+            }
+        }
+
+
+class TaskNodeMetadataAPIResponse(BaseAPIResponse):
+    code: str
+    filepath: str
+    parameters: Dict
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "code": "def split_data(data: pd.DataFrame, parameters: Dict) -> Tuple:",
+                "filepath": "proj/src/new_kedro_project/pipelines/data_science/nodes.py",
+                "parameters": {"test_size": 0.2},
+            }
+        }
+
+
+class DataNodeMetadataAPIResponse(BaseAPIResponse):
+    filepath: str
+    type: str
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "filepath": "/my-kedro-project/data/03_primary/master_table.csv",
+                "type": "kedro.extras.datasets.pandas.csv_dataset.CSVDataSet",
+            }
+        }
+
+
+class ParametersNodeMetadataAPIResponse(BaseAPIResponse):
+    parameters: Dict
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "parameters": {
+                    "test_size": 0.2,
+                    "random_state": 3,
+                    "features": [
+                        "engines",
+                        "passenger_capacity",
+                        "crew",
+                        "d_check_complete",
+                        "moon_clearance_complete",
+                        "iata_approved",
+                        "company_rating",
+                        "review_scores_rating",
+                    ],
+                }
+            }
+        }
+
+
+NodeMetadataAPIResponse = Union[
+    TaskNodeMetadataAPIResponse,
+    DataNodeMetadataAPIResponse,
+    ParametersNodeMetadataAPIResponse,
+]
+
+
+class DataNodeAPIResponse(BaseGraphNodeAPIResponse):
+    layer: Optional[str]
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "id": "d7b83b05",
+                "name": "Master Table",
+                "full_name": "master_table",
+                "tags": [],
+                "pipelines": ["__default__", "dp", "ds"],
+                "modular_pipelines": [],
+                "type": "data",
+                "layer": "primary",
+            }
+        }
+
+
+class GraphEdgeAPIResponse(BaseAPIResponse):
+    source: str
+    target: str
+
+
+class NamedEntityAPIResponse(BaseAPIResponse):

```

Comment:
What does this mean? 😅

Reply:
It's supposed to model an API field that has an ID and a name. Used for representing the list of modular pipelines and pipelines in the API response for now. I can add this to the docstring!

Reply:
Ah lol I did delete this in subsequent commits. I guess I came to my senses.